addons/web/static/src/js/views/calendar/calendar_renderer.js

```diff
@@ -386,6 +386,16 @@ return AbstractRenderer.extend({
         }
         this.trigger_up('openCreate', this._convertEventToFC3Event(data));
     },
+    /**
+     * Get the display time for an event in the same day but not all day
+     *
+     * @param {Object} event is the calendar event
+     * @returns {string} the display time
+     */
+    _getDisplayTimeForSingleDayEvent: function (event) {
+        const start = event.extendedProps.r_start || event.start;
+        return moment(start).clone().format(this._getDbTimeFormat());
```

Comment:
if this is the display time for the user, it should be in user's TZ and locale
Also, in the docstring, when you describe datetimes, and dates, it is always a good idea to specify in what TZ you expect that date(time)